File: Dot examples/ASP Book Examples/Book Examples/Chapter18/VB/ShowEntityBase/App_Code/Entities/Movie.vb
imports System

Partial Public Class Movie
    Inherits EntityBase(Of Movie)

    Protected Overrides Sub Validate()
        ' Title is required
        If Not ValidationUtility.SatisfiesRequired(Title) Then
            ValidationErrors.Add("Title", "Required")
        End If

        ' Director is required
        If Not ValidationUtility.SatisfiesRequired(Director) Then
            ValidationErrors.Add("Director", "Required")
        End If

        ' DateReleased is required
        If DateReleased = DateTime.MinValue Then
            ValidationErrors.Add("DateReleased", "Required")
        End If

        ' DateReleased can't be more than 10 years ago
        If (DateTime.Now.Year - DateReleased.Year) > 10 Then
            ValidationErrors.AddIfNotAlready("DateReleased", "Movie too old")
        End If

    End Sub

End Class

File: Dot examples/ASP Book Examples/Book Examples/Chapter18/CS/ShowEntityBase/App_Code/Entities/Movie.cs
using System;

public partial class Movie : EntityBase<Movie>
{
    protected override void Validate()
    {
        // Title is required
        if (!ValidationUtility.SatisfiesRequired(Title))
            ValidationErrors.Add("Title", "Required");
        // Director is required
        if (!ValidationUtility.SatisfiesRequired(Director))
            ValidationErrors.Add("Director", "Required");
        // DateReleased is required
        if (DateReleased == DateTime.MinValue)
            ValidationErrors.Add("DateReleased", "Required");
        // DateReleased can't be more than 10 years ago
        if ((DateTime.Now.Year - DateReleased.Year) > 10)
            ValidationErrors.AddIfNotAlready("DateReleased", "Movie too old");
    }
}
